component/azstorage/block_blob.go

```diff
@@ -753,7 +753,7 @@ func (bb *BlockBlob) Write(name string, offset, length int64, data []byte, fileO
 			}
 		}
 		// WriteFromBuffer should be able to handle the case where now the block is too big and gets split into multiple blocks
-		err := bb.WriteFromBuffer(name, nil, *dataBuffer)
+		err := bb.WriteFromBuffer(name, metadata, *dataBuffer)
```

Comment:
In stage block and put block list also we need to take care of this metadata update?